include/osw_hal.h

```diff
@@ -181,51 +196,47 @@ class OswHal {
-    void getUTCTime(uint32_t* hour, uint32_t* minute, uint32_t* second);
+    void getUTCTime(OswTime& oswTime);
 
     // Offset getters for primary / secondary time (cached!)
     time_t getTimezoneOffsetPrimary();
     time_t getTimezoneOffsetSecondary();
 
     // New time functions with offset
     time_t getTime(time_t& offset);
-    void getTime(time_t& offset, uint32_t* hour, uint32_t* minute, uint32_t* second, bool* afterNoon = nullptr);
-    void getDate(time_t& offset, uint32_t* day, uint32_t* weekDay);
-    void getDate(time_t& offset, uint32_t* day, uint32_t* month, uint32_t* year);
-    const char* getWeekday(time_t& offset, uint32_t* setWDay = nullptr);
+    void getTime(time_t& offset, OswTime& oswTime);
+    void getDate(time_t& offset, OswDate& oswDate, uint32_t* setWDay = nullptr);
 
     // For backward compatibility: Local time functions (= primary timezone)
-    inline void getLocalTime(uint32_t* hour, uint32_t* minute, uint32_t* second, bool* afterNoon = nullptr) {
-        this->getTime(this->timezoneOffsetPrimary, hour, minute, second, afterNoon);
+    inline void getLocalTime(OswTime& oswTime) {
+        this->getTime(this->timezoneOffsetPrimary, oswTime);
     }
     inline uint32_t getLocalTime() {
         return this->getTime(this->timezoneOffsetPrimary);
     }
-    inline void getLocalDate(uint32_t* day, uint32_t* weekDay) {
-        this->getDate(this->timezoneOffsetPrimary, day, weekDay);
-    };
-    inline void getLocalDate(uint32_t* day, uint32_t* month, uint32_t* year) {
-        this->getDate(this->timezoneOffsetPrimary, day, month, year);
-    };
-    inline const char* getLocalWeekday(uint32_t* sWDay = nullptr) {
-        return this->getWeekday(this->timezoneOffsetPrimary, sWDay);
+    inline void getLocalDate(OswDate& oswDate, uint32_t* sWDay = nullptr) {
+        this->getDate(this->timezoneOffsetPrimary, oswDate, sWDay);
     };
 
     // For backward compatibility: Dual time functions (= secondary timezone)
-    inline void getDualTime(uint32_t* hour, uint32_t* minute, uint32_t* second, bool* afterNoon = nullptr) {
-        this->getTime(this->timezoneOffsetSecondary, hour, minute, second, afterNoon);
+    inline void getDualTime(OswTime& oswTime) {
+        this->getTime(this->timezoneOffsetSecondary, oswTime);
     }
     inline uint32_t getDualTime() {
         return this->getTime(this->timezoneOffsetSecondary);
     }
-    inline void getDualDate(uint32_t* day, uint32_t* weekDay) {
-        this->getDate(this->timezoneOffsetSecondary, day, weekDay);
+    inline void getDualDate(OswDate& oswDate, uint32_t* sWDay = nullptr) {
+        this->getDate(this->timezoneOffsetPrimary, oswDate, sWDay);
     };
-    inline void getDualDate(uint32_t* day, uint32_t* month, uint32_t* year) {
-        this->getDate(this->timezoneOffsetSecondary, day, month, year);
-    };
-    inline const char* getDualWeekday(uint32_t* sWDay = nullptr) {
-        return this->getWeekday(this->timezoneOffsetSecondary, sWDay);
```

Comment:
`For backward compatibility`... I would explicitly not remove these functions for now...

Reply:
There is no place to use the old code.

I've already changed the code from the old version to the new one, should I keep it?

Reply:
Okay... :roll_eyes:

Reply:
```
[[deprecated("Use OswHal::getDualDate instead.")]]
inline void getDualDate(uint32_t* day, uint32_t* month, uint32_t* year) {
    this->getDate(this->timezoneOffsetSecondary, day, month, year);
};
```
maybe this is style?

Reply:
Yes - before just removing stuff, first mark it as deprecated. Especially since the compiler will strip unused, deprecated functions anyway...